src/kvstore/wal/FileBasedWal.cpp

```diff
@@ -546,132 +498,75 @@ bool FileBasedWal::rollbackToLog(LogID id) {
-            (*it)->markInvalid();
-            it = buffers_.erase(it);
-        }
+    // First close the current file
+    closeCurrFile();
 
-        if (!buffers_.empty()) {
-            // Found the target log id
-            buf = buffers_.back();
-            foundTarget = true;
-            lastLogId_ = id;
-            lastLogTerm_ = buf->getTerm(lastLogId_ - buf->firstLogId());
+    {
+        std::lock_guard<std::mutex> g(walFilesMutex_);
 
-            // Create a new buffer starting from id + 1
-            createNewBuffer(id + 1, g);
-            // Since the log id is rolled back, we need to close
-            // the previous wal file
-            buffers_.back()->rollover();
+        if (!walFiles_.empty()) {
+            auto it = walFiles_.upper_bound(id);
+            // We need to remove wal files whose entire log range
+            // are rolled back
+            while (it != walFiles_.end()) {
+                // Need to remove the file
+                VLOG(1) << "Removing file " << it->second->path();
+                unlink(it->second->path());
+                it = walFiles_.erase(it);
+            }
         }
-    }
 
-    if (foundTarget) {
-        CHECK(buf != nullptr);
-       if (buf->freeze()) {
-            // Flush the incomplete buffer which the target log id resides in
-            flusher_->flushBuffer(shared_from_this(), buf);
-        }
-    }
-
-    int fd{-1};
-    while (!foundTarget) {
-        LOG(WARNING) << "Need to rollback from files."
-                        " This is an expensive operation."
-                        " Please make sure it is correct and necessary";
-
-        // Close the current file fist
-        closeCurrFile();
-
-        std::lock_guard<std::mutex> g(walFilesMutex_);
         if (walFiles_.empty()) {
-            CHECK_EQ(id, 0);
-            foundTarget = true;
+            // All WAL files are gone
+            CHECK(id == firstLogId_ - 1 || id == 0);
+            firstLogId_ = 0;
             lastLogId_ = 0;
-            break;
+            lastLogTerm_ = 0;
+        } else {
+            VLOG(1) << "Roll back to log " << id
+                    << ", the last WAL file is now \""
+                    << walFiles_.rbegin()->second->path() << "\"";
+            lastLogId_ = id;
+            lastLogTerm_ = readTermId(walFiles_.rbegin()->second->path(), lastLogId_);
         }
 
-        auto it = walFiles_.upper_bound(id);
+        // Create the next WAL file
+        prepareNewFile(lastLogId_ + 1);
+    }
 
-        // We need to remove wal files whose entire log range
-        // are rolled back
-        while (it != walFiles_.end()) {
-            // Need to remove the file
-            VLOG(2) << "Removing file " << it->second->path();
-            unlink(it->second->path());
-            it = walFiles_.erase(it);
-        }
+    //------------------------------
+    // 2. Roll back in-memory buffers
+    //------------------------------
+    {
```

Comment:
We don't need rollback the buffers.   Just clean them all.

Reply:
Yes, cleaning all of them is simple, but it might hurt the perf. For example, we have to read from the file